fdo.go

```diff
@@ -0,0 +1,264 @@
+package desktop
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"fyne.io/fyne"
+)
+
+var iconExtensions = []string{".png", ".svg", ".xpm"}
+
+//FdoDesktop contains the information from Linux .desktop files
+type FdoDesktop struct {
+	Name     string
+	IconName string
+	IconPath string
+	Exec     string
+}
+
+//FdoResourceFormat turns a string into a usable name for fyne.Resources
+func FdoResourceFormat(name string) string {
```

Comment:
Perhaps this is "ResourceName" as it's not really a format.
Aditionally would it not be better to encapsulate this privately and expose a "LoadIcon() fyne.Resource" method to avoid duplicated use of this internal detail?